foundation/src/main/java/org/cbio/portal/pipelines/foundation/model/staging/MutationData.java

```diff
@@ -110,23 +110,27 @@ public MutationData(String sampleId, ShortVariantType shortVariant) {
         String[] alleleList = bases.toUpperCase().split(" ");
         
         this.refAllele = FoundationUtils.resolveReferenceAllele(bases, alleleList, shortVariant.getCdsEffect(), shortVariant.getStrand());
-        if (this.refAllele.isEmpty()) {
+        if (refAllele.isEmpty() && shortVariant.getCdsEffect().contains("del")) {
```

Comment:
Why the mixture of this.refAllele and refAllele?  I don't see  a local refAllele.  This is true for other members as well, for example variantType